benchmark/scatter.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+const fork = require('child_process').fork;
+const path = require('path');
+const CLI = require('./_cli.js');
+
+//
+// Parse arguments
+//
+const cli = CLI(`usage: ./node benchmark/scatter.js <filename>
+  Run the benchmark script <filename> many times and output the rate (ops/s)
+  together with the benchmark variables as a csv.
+
+  --runs 30              number of samples
+  --set  variable=value  set benchmark variable (can be repeated)
+`, {
+  arrayArgs: ['set']
+});
+
+if (cli.required.length !== 1) {
+  cli.abort(cli.usage);
+  return;
+}
+
+// Create queue from the benchmarks list such both node versions are tested
+// `runs` amount of times each.
+const filepath = path.resolve(cli.required[0]);
+const name = filepath.slice(__dirname.length + 1);
+const runs = cli.optional.runs ? parseInt(cli.optional.runs, 10) : 30;
+
+let printHeader = true;
+
+(function recursive(i) {
+  const child = fork(path.resolve(__dirname, filepath), cli.optional.set);
+
+  child.on('message', function(data) {
+    // print csv header
+    if (printHeader) {
+      const confHeader = Object.keys(data.conf).join('", "');
+      console.log(`"filename", "${confHeader}", "rate", "time"`);
+      printHeader = false;
+    }
+
+    // print data row
+    const confData = Object.keys(data.conf)
+      .map((key) => data.conf[key])
+      .join(', ');
```

Comment:
This may need to be improved a bit as far as escaping goes, at least in case there are quotes in any of the inputs. I'm not sure how R parses CSV, but it may also have issues with other characters, including control characters (especially CR and LF which may be used in some benchmarks -- namely http).